File: src/main/java/com/task/interview/colliers/digital/utils/DateUtils.java
package com.task.interview.colliers.digital.utils;

import com.task.interview.colliers.digital.exception.InvalidDateParametersException;
import lombok.extern.slf4j.Slf4j;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

@Slf4j
public class DateUtils {

    public static Date createDateWithFormat(String date, String format) {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(format);
        try {
            return simpleDateFormat.parse(date);
        } catch (ParseException e) {
            log.error("Cannot parse date: " + date + " with format: " + format);
            throw new InvalidDateParametersException();
        }
    }

}

File: src/main/java/com/task/interview/colliers/digital/utils/DateUtils.kt
package com.task.interview.colliers.digital.utils

import com.task.interview.colliers.digital.exception.InvalidDateParametersException
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {

    @JvmStatic
    fun createDateWithFormat(date: String, format: String): Date {
        val simpleDateFormat = SimpleDateFormat(format)
        return try {
            simpleDateFormat.parse(date)
        } catch (e: ParseException) {
            throw InvalidDateParametersException()
        }
    }

}